controllers/glance_controller.go

```diff
@@ -1192,3 +1195,26 @@ func (r *GlanceReconciler) ensureDB(
 	instance.Status.Conditions.MarkTrue(condition.DBReadyCondition, condition.DBReadyMessage)
 	return db, ctrlResult, nil
 }
+
+// checkGlanceAPIsGeneration -
+func (r *GlanceReconciler) checkGlanceAPIsGeneration(
+	ctx context.Context,
+	instance *glancev1.Glance,
+) (bool, error) {
+	cond := false
+	// get all GlanceAPI CRs
+	glances := &glancev1.GlanceAPIList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(instance.Namespace),
+	}
+	if err := r.Client.List(context.Background(), glances, listOpts...); err != nil {
+		r.Log.Error(err, "Unable to retrieve Glance CRs %w")
+		return cond, err
+	}
+	for _, item := range glances.Items {
+		if item.Generation == item.Status.ObservedGeneration {
+			cond = true
```

Comment:
you probably want to return true only when all the glanceAPIs are up-to-date. But this code now returns true if any of the GlanceAPI is up-to-date.

I think something like:
```golang
	for _, item := range glances.Items {
		if item.Generation != item.Status.ObservedGeneration {
			return false, nil
        }
    }
    return true, nil
```

Reply:
yeah, I think it's more efficient always return false if one of the API is not up-to-date.

Reply:
It not just about efficiency as the original code would return true if one GlanceAPI is up-to-date but the other GanceAPI is not up-to-date. My suggested code returns false in that case.